CMakeLists.txt

```diff
@@ -1,86 +1,53 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(control_toolbox)
 
-# Load catkin and all dependencies required for this package
-find_package(catkin REQUIRED COMPONENTS
-  cmake_modules
-  control_msgs
-  dynamic_reconfigure
-  message_generation
-  realtime_tools
-  roscpp
-  std_msgs
-  )
-
-find_package(Boost REQUIRED COMPONENTS system thread)
-find_package(TinyXML REQUIRED)
-
-# Dynamics reconfigure
-generate_dynamic_reconfigure_options(
-  cfg/Parameters.cfg
-  )
-
-# Add services and generate them
-add_service_files(
-  FILES
-  SetPidGains.srv
-  )
-
-generate_messages(
-  DEPENDENCIES std_msgs
-)
-
-# Declare catkin package
-catkin_package(
-  DEPENDS TinyXML
-  CATKIN_DEPENDS
-    control_msgs
-    dynamic_reconfigure
-    message_runtime
-    realtime_tools
-    roscpp
-    std_msgs
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  )
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
-  ${Boost_INCLUDE_DIR}
-  ${TinyXML_INCLUDE_DIRS}
-  )
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(realtime_tools REQUIRED)
 
 add_library(${PROJECT_NAME}
   src/dither.cpp
   src/limited_proxy.cpp
   src/pid.cpp
-  src/pid_gains_setter.cpp
+  # src/pid_gains_setter.cpp
   src/sine_sweep.cpp
   src/sinusoid.cpp
   )
-add_dependencies(${PROJECT_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+ament_target_dependencies(${PROJECT_NAME} "rclcpp")
+ament_target_dependencies(${PROJECT_NAME} "realtime_tools")
+target_include_directories(${PROJECT_NAME} PUBLIC include)
 
-target_link_libraries(${PROJECT_NAME} ${TinyXML_LIBRARIES} ${catkin_LIBRARIES} ${Boost_LIBRARIES})
+if(BUILD_TESTING)
+  find_package(ament_cmake_gmock REQUIRED)
+  find_package(ament_cmake_gtest REQUIRED)
```

Comment:
gtest < gmock AFAIK, you could use the gmock macro for all these tests and should be ok

Reply:
Since there are both gtest and gmock tests I rather have both if it is all the same to you.